### DISCUSSION TITLE
What "Type" to use in systemd?

### DISCUSSION THREAD

**Discussion by Ra2-IFV** [transmission/transmission#4768] [Q&A] [answered]
Here's the thing, I want to create a transmission systemd service, but I'm confused about the “type” of service to choose.
I used the configuration file from the [Ubuntu deb package](https://packages.ubuntu.com/kinetic/transmission-daemon):
```
[Unit]
Description=Transmission BitTorrent Daemon
After=network.target

[Service]
User=debian-transmission
Type=notify
ExecStart=/usr/bin/transmission-daemon -f --log-error
ExecStop=/bin/kill -s STOP $MAINPID
ExecReload=/bin/kill -s HUP $MAINPID
NoNewPrivileges=true

[Install]
WantedBy=multi-user.target
```
This service file works fine on transmission 3.0 (actually, 3.00-2ubuntu2), but it doesn't work on transmission 4.0.
I got this error from systemd:
```
Feb 09 17:35:52 HOSTNAME systemd[1]: transmission-daemon.service: start operation timed out. Terminating..
Feb 09 17:35:52 HOSTNAME systemd[1]: transmission-daemon.service: Failed with result 'timeout'.
```
Actually, Transmission did start successfully, but it doesn't seem to notify systemd when it's done, so systemd killed it.
After I changed `Type` to `simple`, this configuration file works again.
So my question is, since `Type=notify` no longer works on transmission 4, which type should I use?

**Comment by jcpvdm** [comment, marked as answer]
I had the same problem.

When running `cmake `with default arguments, it scans the system for SYSTEMD support. if does not find it, it assumes SYSTEMD is not installed and compiles transmission-daemon without support for notify. You can confirm this by running `cmake `with explicit argument "-DWITH_SYSTEMD=ON" (default is AUTO). It should give an error that it could not find SYSTEMD and `cmake ` then aborts. 
`cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DWITH_SYSTEMD=ON ..`

This is resolved by installing package libsystemd-dev.
`sudo apt install libsystemd-dev`

After that, the above `cmake ` command completes without error. Then just run `make` again, and should compile transmission-daemon with support for notify.
